### PR TITLE
crud food is working

### PR DESCRIPTION
**Pull request by hasandk**

### CODE REVIEW

File: src/main/java/com/base/site/controllers/FoodController.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        log.info("  get mapping food is called");
+
+            return "food";
+    }
+
+    @GetMapping("/createFood")
+    public String createStudentForm(Model model) {
+
+            Food food = new Food();
+            model.addAttribute("food", food);
```
Comment (mrtauern) [inline src/main/java/com/base/site/controllers/FoodController.java:39]: TIP: You can also write it like this
`Food food = model.addAttribute("food", food);`
Then we save some line of code 😉

File: src/main/java/com/base/site/repositories/FoodRepo.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.base.site.repositories;
+
+import com.base.site.models.Food;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository ("FoodRepo")
+public interface FoodRepo extends JpaRepository<Food, Integer> {
```
Comment (mrtauern) [inline src/main/java/com/base/site/repositories/FoodRepo.java]: We use Long and not int or integer in our project for DB ID's

File: src/main/java/com/base/site/models/Food.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Food {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
```
Comment (mrtauern) [inline src/main/java/com/base/site/models/Food.java]: We use Long and not int or integer in our project for DB ID's

File: src/main/java/com/base/site/security/FoodService.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import java.util.List;
+
+@Service("FoodService")
+public interface FoodService {
+
+    List<Food> getAllFood();
+    Food saveFood(Food food);
+    Food getFoodById(int id);
+    void deleteFoodById(int id);
```
Comment (mrtauern) [inline src/main/java/com/base/site/security/FoodService.java]: Use findAll(), findById(), save(), deleteById(), and delete() so we have one standard across our project.

File: src/main/resources/templates/createFood.html [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<html lang="en" xmlns="http://www.w3.org/1999/xhtml"
+      xmlns:th="http://www.thymeleaf.org">
+<head>
+  <meta charset="UTF-8">
+  <title>Add Food</title>
+</head>
```
Comment (mrtauern) [inline src/main/resources/templates/createFood.html]: Replace this with this:
`<head th:include="fragments/head :: styles"></head>`

File: src/main/resources/templates/fragments/navbar_top.html [hunk 6]
```diff
@@ -9,7 +9,8 @@
             <!-- <label for="checkbox_toggle" class="hamburger">&#9776;</label> -->
             <!-- NAVIGATION MENUS -->
             <div class="menu">
-                <li><a href="/">Home</a></li>
+                <li><a href="/food">Food</a></li>
+                <li><a href="/index">Home</a></li>
```
Comment (mrtauern) [inline src/main/resources/templates/fragments/navbar_top.html]: We have to find a standard to how we do this

File: src/main/java/com/base/site/controllers/FoodController.java [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+            return "createFood";
+    }
+
+    @PostMapping("/saveFood")
+    public String saveFood(@ModelAttribute("food") Food food, Model model) {
+            foodService.saveFood(food);
+            log.info("  PostMapping saveFood is called ");
+
+            return  "redirect:/" + "food";
+    }
```
Comment (mrtauern) [inline src/main/java/com/base/site/controllers/FoodController.java]: Is this used for both create and update food?

File: src/main/java/com/base/site/controllers/FoodController.java [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+    @GetMapping("/showUpdateFood/{id}")
```
Comment (mrtauern) [inline src/main/java/com/base/site/controllers/FoodController.java]: Can we just call it updateFood 😉

**Review by mrtauern — CHANGES_REQUESTED** [review]
It looks very good, but there are some minor issues that have to be fixed 😉

File: src/main/java/com/base/site/models/Food.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+    public void setProtein(int protein) {
+        this.protein = protein;
+    }
+
+    public int getCarbohydrates() {
+        return carbohydrates;
+    }
+
+    public void setCarbohydrates(int carbohydrates) {
```
Comment (niklasld) [inline src/main/java/com/base/site/models/Food.java]: Database carbohydrates, fat and protein have all been changed to doubles

File: src/main/java/com/base/site/models/Food.java [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+@Entity
+public class Food {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+    private String name;
+    private int protein;
+    private int carbohydrates;
+    private int fat;
```
Comment (niklasld) [inline src/main/java/com/base/site/models/Food.java]: Database carbohydrates, fat and protein have all been changed to doubles

File: src/main/resources/templates/fragments/navbar_top.html (hunk 6 above)
Comment (mrtauern) [inline src/main/resources/templates/fragments/navbar_top.html]: Is the link to our home page "/index"?

**Review by mrtauern — APPROVED** [review]
It looks fine

**Pull request merged**